pandas/core/frame.py

```diff
@@ -703,14 +703,15 @@ def itertuples(self, index=True, name="Pandas"):
 
         # Python 3 supports at most 255 arguments to constructor, and
         # things get slow with this many fields in Python 2
-        if len(self.columns) + index < 256:
+        if name is not None and len(self.columns) + index < 256:
             # `rename` is unsupported in Python 2.6
             try:
                 itertuple = collections.namedtuple(
                     name, fields+list(self.columns), rename=True)
-                return (itertuple(*row) for row in zip(*arrays))
-            except:
+            except Exception:
                 pass
+            else:
```

Comment:
the statement you added duplicates the other return statement. you don't need the else clause at all. if you think that the ``_make`` actually speeds things, then put it there. 

- would need a benchmark as well
- need a test for ``name=None`` if this actually changed anything (does it?)